openpgp/packet/private_key.go

```diff
@@ -217,7 +276,7 @@ func (pk *PrivateKey) Serialize(w io.Writer) (err error) {
 		return
 	}
 
-	if len(privateKeyBytes) > 0 {
+	if len(privateKeyBytes) > 0 && !pk.Encrypted {
```

Comment:
This branch is not taken when `pk.Encrypted` because the checksum is already in `privateKeyBytes` (in encrypted form), right?